bullet_train/app/models/concerns/memberships/base.rb

```diff
@@ -30,9 +30,9 @@ module Memberships::Base
       end
     end
 
-    after_create_commit :publish_changed_quantity
-    after_update_commit :publish_changed_quantity
-    after_destroy_commit :publish_changed_quantity
+    after_create :publish_changed_quantity
+    after_update :publish_changed_quantity
+    after_destroy :publish_changed_quantity
```

Comment:
@julianrubisch this is really weird. I just ran some tests and the callback for `after_destroy_commit` is firing, but not the other two. I tried a generic `after_save_commit` and that didn't fire either. But just a plain `after_commit` _does_ fire. So I think we could replace all three callbacks with a single `after_commit` and we'd get what we want.

```suggestion
    after_commit :publish_changed_quantity
```

Reply:
Looks like this is the problem: https://github.com/rails/rails/issues/29554

Each of the three `after_*_commit` callbacks are just aliases for `after_commit on: :*` so whichever is declared last wins. For some reason the Rails team doesn't see this as being a bug.

Reply:
Oh my, now I **remember**! Seems like every 5 years I have to stumble upon this ...

Reply:
Yeah, I feel like I've run into this before as well, and never seem to remember. To me it seems like an _obvious_ problem that should be fixed in Rails itself. Not sure why the Rails teams doesn't think so.